timesketch/api/v1/resources/explore.py

```diff
@@ -156,15 +156,23 @@ def post(self, sketch_id: int):
 
         all_indices = list({t.searchindex.index_name for t in sketch.timelines})
         indices = query_filter.get("indices", all_indices)
+        all_timeline_ids = [t.id for t in sketch.timelines]
+        timeline_ids = query_filter.get("timelines", all_timeline_ids)
 
         # If _all in indices then execute the query on all indices
         if "_all" in indices:
             indices = all_indices
 
         # Make sure that the indices in the filter are part of the sketch.
-        # This will also remove any deleted timeline from the search result.
         indices, timeline_ids = get_validated_indices(indices, sketch)
-
+        # the timeline_ids seem to be [] all the time, make a check if it its empty
+        if not timeline_ids:
+            timeline_ids = all_timeline_ids
+        if not timeline_ids:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "No valid timeline ids were found to perform the search on.",
+            )
```

Comment:
I think we can keep this. It does ensure an error in case we don't find a timeline id.